eternalcore-core/src/main/java/com/eternalcode/core/feature/chat/ChatSoundController.java

```diff
@@ -0,0 +1,28 @@
+package com.eternalcode.core.feature.chat;
+
+import com.eternalcode.core.injector.annotations.component.Controller;
+import com.eternalcode.core.notice.NoticeService;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.AsyncPlayerChatEvent;
+
+@Controller
+class ChatSoundController implements Listener {
+
+    private final NoticeService noticeService;
+    private final ChatSettings settings;
+
+    ChatSoundController(NoticeService noticeService, ChatSettings settings) {
+        this.noticeService = noticeService;
+        this.settings = settings;
+    }
+
+    @EventHandler
+    void onSound(AsyncPlayerChatEvent event) {
```

Comment:
```suggestion
    void onChat(AsyncPlayerChatEvent event) {
```
We are listening when the player chats, not when the player makes a sound. Therefore, the appropriate name for this listener is `onChat`.